CollAction/Frontend/app/donation/DebitDetailsBox.tsx

```diff
@@ -0,0 +1,128 @@
+﻿import * as React from "react";
+import { IdealBankElement, IbanElement } from 'react-stripe-elements';
+
+interface IDebitDetailsBoxProps {
+    stripe: any;
+    amount: number;
+    userEmail: string;
+    userName: string;
+    isRecurring: boolean;
+}
+
+interface IDebitDetailsBoxState {
+    showError: boolean;
+    error: string;
+}
+
+export default class DebitDetailsBox extends React.Component<IDebitDetailsBoxProps, IDebitDetailsBoxState> {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showError: false,
+            error: ""
+        };
+    }
+
+    submitPayment(event: React.FormEvent<HTMLFormElement>) : Promise<void> {
+        event.preventDefault();
+        this.setState({ showError: false });
+
+        if (this.props.isRecurring) {
+            return this.submitPaymentSepa();
+        }
+        else {
+            return this.submitPaymentIdeal();
+        }
+    }
+
+    async submitPaymentSepa(): Promise<void> {
+        let sourceData = {
+            type: 'sepa_debit',
+            currency: 'eur',
+            owner: {
+                name: this.props.userName,
+                email: this.props.userEmail
+            },
+            mandate: {
+                notification_method: 'email' // Automatically send a mandate notification email to your customer once the source is charged
+            }
+        };
+
+        let response = await this.props.stripe.createSource(sourceData);
+        if (response.error) {
+            console.log("Unable to start SEPA Direct: " + response.error);
+            this.setState({ showError: true, error: "Unable to start SEPA Direct" })
```

Comment:
Since this message is shown to the user, maybe a more user-oriented text would be better?

Reply:
I'm not sure what could be better, if you have suggestions... This will only show if there's a bug, or an issue on stripe's end, so should be rare.

Reply:
I added more user-friendly messages, feel free to improve them if you have ideas.